### PR TITLE
Standardize usage of log.exception

### PR DESCRIPTION
**Pull request by dannon**
Standardizes our usage of log.exception to 1) not include the exception in the string, and 2) use log.exception parameters for string templating.

### CODE REVIEW

**Comment by nsoranzo** [comment]
Thanks! By the way, the `str()` castings should be unnecessary.

**Comment by dannon** [comment]
Yep, also going to convert usage of `log.exception("oops %s" % var)` to `log.exception("oops %s", var)`

(going to make multiple passes to make sure I get each class of error)

**Comment by nsoranzo** [comment]
Fantastic! I was going to suggest it, but I didn't want to be too nitpicky!

**Comment by nsoranzo** [comment]
Also take a look at the flake8 errors, many exception variables are now unused.

**Comment by martenson** [comment]
@dannon this has both WIP and review labels

File: lib/galaxy/webapps/galaxy/api/lda_datasets.py [hunk 1]
```diff
@@ -607,7 +607,7 @@ def traverse( folder ):
                 log.exception( "Unable to create archive for download" )
                 raise exceptions.InternalServerError( "Unable to create archive for download." )
             except Exception:
-                log.exception( "Unexpected error %s in create archive for download" % sys.exc_info()[ 0 ] )
+                log.exception( "Unexpected error %s in create archive for download", sys.exc_info()[ 0 ] )
```
Comment (nsoranzo) [inline lib/galaxy/webapps/galaxy/api/lda_datasets.py:610]: This one could probably be refactored even more to just:
```
log.exception("Unexpected error in create archive for download")
```

File: lib/galaxy/webapps/galaxy/controllers/library_common.py [hunk 2]
```diff
@@ -1863,7 +1863,7 @@ def __str__( self ):
                     status = 'error'
                 except:
                     error = True
-                    log.exception( "Unexpected error %s in create archive for download" % sys.exc_info()[0] )
+                    log.exception( "Unexpected error %s in create archive for download", sys.exc_info()[0] )
```
Comment (nsoranzo) [inline lib/galaxy/webapps/galaxy/controllers/library_common.py:1866]: Also this one could probably be refactored even more to just:
```
log.exception("Unexpected error in create archive for download")
```

File: lib/tool_shed/galaxy_install/tools/tool_panel_manager.py [hunk 3]
```diff
@@ -89,8 +89,8 @@ def config_elems_to_xml_file( self, config_elems, config_filename, tool_path ):
                 fh.write( xml_util.xml_to_string( elem, use_indent=True ) )
             fh.write( '</toolbox>\n' )
             fh.close()
-        except Exception as e:
-            log.exception( "Exception in ToolPanelManager.config_elems_to_xml_file: %s" % str( e ) )
+        except Exception:
+            log.exception( "Exception in ToolPanelManager.config_elems_to_xml_file: %s" )
```
Comment (nsoranzo) [inline lib/tool_shed/galaxy_install/tools/tool_panel_manager.py:93]: Need to remove `: %s` here.

File: lib/tool_shed/repository_registry.py [hunk 4]
```diff
@@ -341,11 +341,11 @@ def remove_entry( self, repository ):
                 self.unload_repository_and_suite_tuple( repository )
                 if is_level_one_certified:
                     self.unload_certified_level_one_repository_and_suite_tuple( repository )
-        except Exception as e:
+        except Exception:
             # The viewable repository numbers and the categorized (filtered) lists of repository tuples
             # may be slightly skewed, but that is no reason to result in a potential server error.  All
             # will be corrected at next server start.
-            log.exception( "Handled error removing entry from repository registry: %s." % str( e ) )
+            log.exception( "Handled error removing entry from repository registry: %s." )
```
Comment (nsoranzo) [inline lib/tool_shed/repository_registry.py:348]: Need to remove `: %s.` here.

**Review by nsoranzo — COMMENTED** [review]
Sorry for late review, I didn't get notification for the label change.

**Comment by martenson** [comment]
@nsoranzo label changes do not notify afaik :'(

**Comment by dannon** [comment]
@nsoranzo Thanks for finding these issues, opening PR fixing it now.

**Pull request merged**